numpy/core/_methods.py

```diff
@@ -191,6 +191,17 @@ def _var(a, axis=None, dtype=None, out=None, ddof=0, keepdims=False):
     x = asanyarray(arr - arrmean)
     if issubclass(arr.dtype.type, (nt.floating, nt.integer)):
         x = um.multiply(x, x, out=x)
+    # Fast-paths for complex types - avoid conjugate
+    elif arr.dtype.type is nt.complex128:
+        xv = x.view(dtype=(nt.float64, (2,)))
+        um.multiply(xv, xv, out=xv)
+        x = um.add(xv[..., 0], xv[..., 1], out=x.real).real
+    elif arr.dtype.type is nt.complex64:
```

Comment:
Can you add `clongdouble` too? Perhaps add a dictionary to map the types.

Reply:
Agreed, one thing to take care of (and that should have a test) is that non-native byte order does not use the view, or only uses it with the correct byte order. (The identity check does not catch byte order differences. An equality check, even if a bit strange maybe, does). Not sure of the top of my head how best to write it nicest.
After my PR goes in, we could even think about adding a `.real_dtype()` method ;).

Reply:
Thanks @eric-wieser & @seberg for the suggestions! I've made the following changes/updates:

 - Added a fast-path for `complex256`.
 - Re-formulated the complex type checking to use a dict (eliminates some LOC's, doesn't appear to hurt the performance)
 - Added an additional check for the byteorder - any array that has anything other than non-native byte order will be funnelled to the original "slow" path so that it doesn't hit the `view`s.
 - Updated the tests:
   * Modified a test to check `complex{64,128,256}` input types
   * Added a test to verify `var()` gives the same result for complex arrays with both native and non-native byte order.

Let me know what you think of this iteration, and thanks again for the feedback!